airbyte-integrations/connectors/source-salesforce/unit_tests/unit_test.py

```diff
@@ -298,3 +298,30 @@ def test_download_data_filter_null_bytes(stream_bulk_config, stream_bulk_api):
         m.register_uri("GET", f"{job_full_url}/results", content=b'"Id","IsDeleted"\n\x00"0014W000027f6UwQAI","false"\n\x00\x00')
         res = list(stream.download_data(url=job_full_url))
         assert res == [(1, {"Id": "0014W000027f6UwQAI", "IsDeleted": "false"})]
+
+
+@pytest.mark.parametrize(
+    "streams_criteria,number_of_filtered_streams",
+    [
+        ([{"criteria": "exacts", "value": "FakeStream30"}], 1),
+        ([{"criteria": "not exacts", "value": "Fakestream30"}], 99),
+        ([{"criteria": "starts with", "value": "fakeStream3"}], 11),
+        ([{"criteria": "starts not with", "value": "mystream"}], 100),
+        ([{"criteria": "ends with", "value": "stream10"}], 1),
+        ([{"criteria": "ends not with", "value": "0"}], 90),
+        ([{"criteria": "contains", "value": "keStr"}], 100),
+        ([{"criteria": "contains", "value": "eam10"}], 1),
+        ([{"criteria": "not contains", "value": "string"}], 100),
+        ([{"criteria": "not contains", "value": "Mystream"}], 100),
+        ([{"criteria": "not contains", "value": "stream23"}], 99),
+    ],
+)
+def test_discover_with_streams_criteria_param(streams_criteria, number_of_filtered_streams, stream_rest_config):
+    updated_config = {**stream_rest_config, **{"streams_criteria": streams_criteria}}
+    sf_object = Salesforce(**stream_rest_config)
+    sf_object.login = Mock()
+    sf_object.access_token = Mock()
+    sf_object.instance_url = "https://fase-account.salesforce.com"
+    sf_object.describe = Mock(return_value={"sobjects": [{"name": f"FakeStream{i}"} for i in range(100)]})
+    filtered_streams = sf_object.get_validated_streams(config=updated_config)
+    assert len(filtered_streams) == number_of_filtered_streams
```

Comment:
perhaps we may check the steams by name instead of just count them?

Reply:
That is, you propose to use the same method to filter the names of streams, and in the end compare them with each other, it is strange?
Could you tell us in more detail how you see this implementation?

Reply:
I just suggest to compare `filtered_stream == expected_filtered_streams`.
E.g. `['stream1', 'stream2'] == ['stream1']`
Is it possible?